### DISCUSSION TITLE
SignalR: How to forcefully disconnect client from hub?

### DISCUSSION THREAD

**Discussion by Thijmen** [dotnet/aspnetcore#59271] [Q&A]
Hi there,

So, I am trying to disconnect a client from the server. All the examples I've seen send a Disconnect message to the client first, but I don't want to trust clients on doing so. Therefore, I would like to disconnect a client by connection ID. Is there any way to do so, without having to rely too much on reflection?

**Comment by Thijmen** [comment]
I think the best shot here would be creating my own HubLifetimeManager, preferably extending RedisHubLifetimeManager, and then keep track of the connections and inject the HubLifetimeManager at places I want to disconnect, and call a method on my own manager. I am not sure if this is the proper way, though.

**Reply by BrennanConroy** [reply]
https://github.com/dotnet/aspnetcore/issues/34481

**Reply by Thijmen** [reply]
I can work with the arguments that you perhaps don't want this in the API, indeed. The documentation for such use cases lacks developer experience in my opinion. The amount of times that I've seen "just send a disconnect message and hope that the client does this" is unfortunately more than "hey, try to implement your own manager, you should be able to get this in". I've had to dig through the Redis backplane code to wrap my head around it.

Don't get me wrong, I love reading other code - but in this case I hoped there would be some documentation around the subject. Thanks for your efforts on SignalR though.
